src/python/build_management/build_manager.py

```diff
@@ -1111,84 +1151,88 @@ def get_revisions_list(bucket_path, testcase=None):
-    sym_debug_builds_urls = get_build_urls_list(sym_debug_build_bucket_path)
-    if not sym_debug_builds_urls:
-      logs.log_error(
-          'Error getting list of symbolized debug build urls from %s.' %
-          sym_debug_build_bucket_path)
-      return None
-    other_build_url_lists.append((sym_debug_build_bucket_path,
-                                  sym_debug_builds_urls))
+  main_build_urls = build_urls[0]
+  other_build_urls = build_urls[1:]
+  base_build_dir = _base_build_dir(main_build_urls['bucket_path'])
+  _setup_build_directories(base_build_dir)
 
   revision_pattern = revisions.revision_pattern_from_build_bucket_path(
-      release_build_bucket_path)
+      main_build_urls['bucket_path'])
   found_revision = False
-  for release_build_url in release_build_urls:
-    match = re.match(revision_pattern, release_build_url)
+  for build_url in main_build_urls['urls_list']:
+    match = re.match(revision_pattern, build_url)
     if not match:
       continue
 
     revision = revisions.convert_revision_to_integer(match.group(1))
-    if (not other_build_url_lists or all(
-        revisions.find_build_url(l[0], l[1], revision)
-        for l in other_build_url_lists)):
+    if (not other_build_urls or all(
+        revisions.find_build_url(url['bucket_path'], url['urls_list'], revision)
+        for url in other_build_urls)):
       found_revision = True
       break
 
   if not found_revision:
     logs.log_error('Unable to find a matching revision.')
     return None
 
-  return setup_regular_build(revision)
+  build = setup_regular_build(revision, main_build_urls['bucket_path'],
+                              build_prefix)
+  if not build:
+    logs.log_error('Failed to set up a build.')
+    return None
+
+  return build
 
 
-def setup_regular_build(revision):
+def setup_regular_build(revision, bucket_path=None, build_prefix=''):
   """Sets up build with a particular revision."""
-  release_build_bucket_path = environment.get_value('RELEASE_BUILD_BUCKET_PATH')
-  release_build_urls = get_build_urls_list(release_build_bucket_path)
+  if not bucket_path:
+    # Bucket path can be customized, otherwise get it from the default env var.
+    bucket_path = environment.get_value('RELEASE_BUILD_BUCKET_PATH')
+
+  build_urls = get_build_urls_list(bucket_path)
   job_type = environment.get_value('JOB_NAME')
-  if not release_build_urls:
-    logs.log_error('Error getting release build urls for job %s.' % job_type)
+  if not build_urls:
+    logs.log_error('Error getting build urls for job %s.' % job_type)
     return None
 
-  release_build_url = revisions.find_build_url(release_build_bucket_path,
-                                               release_build_urls, revision)
-  if not release_build_url:
+  build_url = revisions.find_build_url(bucket_path, build_urls, revision)
+  if not build_url:
     logs.log_error(
         'Error getting build url for job %s (r%d).' % (job_type, revision))
 
+    if build_prefix:
```

Comment:
why is this condition needed here?

Reply:
Let me try this (and something else) in a separate CL.